File: BE/PrePay/src/main/java/com/d111/PrePay/dto/respond/PublicTeamsRes.java
package com.d111.PrePay.dto.respond;

import com.d111.PrePay.model.Team;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class PublicTeamsRes {
    private long teamId;
    private String teamName;
    private String teamMessage;
    private int teamBalance;
    private String teamInitializerNickname;
    private boolean isLike;

    public PublicTeamsRes(Team team) {
        this.teamName = team.getTeamName();
        this.teamId = team.getId();
        this.teamMessage = team.getTeamMessage();
        this.teamBalance = team.getTeamBalance();
    }
}

File: Prepay/app/src/main/java/com/example/prepay/data/response/PublicTeamsRes.kt
package com.example.prepay.data.response

import com.example.prepay.ui.GroupSearch.Searchable

data class PublicTeamsRes(
    val teamId: Int,
    val teamName: String,
    val teamMessage: String,
    val teamBalance: Int,
    val teamInitializerNickname: String,
    val like: Boolean,
    val imgUrl: String,
    val address: String,
    val distance : Double
) : Searchable {
    override val searchableText : String
        get() = "$teamName"
}